### ISSUE TITLE
responsePath not working correctly

### ISSUE THREAD

**Issue by cfjedimaster** [strongloop/loopback-connector-rest#54]
As far as I can see, responsePath is not working correctly. Given this JSON (http://swapi.co/api/starships/?format=json), I would expect $.results to return an array of results. However, I'm getting a 2D array where result[0] is the array of results. I tested the same JSON against a browser-based version of JSONPath and $.results worked as expected.

**Comment by raymondfeng** [comment]
@cfjedimaster Do you want to test the JSON doc with JSONPath nodejs module? We simply evaluate the `body` using the provided responsePath. See https://github.com/strongloop/loopback-connector-rest/blob/master/lib/rest-builder.js#L546

**Comment by cfjedimaster** [comment]
I'd expect just an array. As an example, check out my online tester for JSONPath here: http://static.raymondcamden.com/demos/2016/03/15/. If you test with the JSON provided, LoopBack returns [[STUFF HERE]] while the JSONPath library by itself is returning [STUFF HERE]. Does that make sense?

**Comment by raymondfeng** [comment]
@cfjedimaster JSONPath by default wrap the result into an array as the path can match multiple entries potentially. There is an option to return the unwrapped result but it's going to be breaking change.

**Comment by cfjedimaster** [comment]
So I ran into this myself w/ JSONPath where some paths, like $.author[*].name, could emit N responses. See my bug report here: https://github.com/s3u/JSONPath/issues/68 

I think LoopBack could handle it the same way - such that you *always* get the result back, period.

I don't know honestly - I've looked at this feature for a grand total of 2 hours, so I'm pretty new, but I can say that having [[x]] as a result just seems a bit wonky. :)
